src/actions/users.js

```diff
@@ -224,6 +224,8 @@ export function getProfiles(offset, limit = General.PROFILE_CHUNK_SIZE) {
             return null;
         }
 
+        const profiles = profileListToMap(profilesList);
```

Comment:
What about two different action types RECEIVED_PROFILES for when you have the map and RECEIVED_PROFILES_LIST which will convert the list to a map?